databroker/experimental/server_ext.py

```diff
@@ -0,0 +1,501 @@
+                "$set": {
+                    "data_url": f"file://localhost/{str(path).replace(os.sep, '/')}"
+                }
+            },
+        )
+
+
+class WritingDataFrameAdapter:
+    structure_family = "dataframe"
+
+    def __init__(self, collection, directory, doc):
+        self.collection = collection
+        self.directory = directory
+        self.doc = Document(**doc)
+        self.dataframe_adapter = None
+
+        if self.doc.data_url is not None:
+            path = self.doc.data_url.path
+            if platform == "win32" and path[0] == "/":
+                path = path[1:]
+
+            self.dataframe_adapter = DataFrameAdapter(
+                dask.dataframe.from_pandas(
+                    pandas.read_parquet(path),
+                    npartitions=self.doc.structure.macro.npartitions,
+                )
+            )
+
+        elif self.doc.data_blob is not None:
+            self.dataframe_adapter = DataFrameAdapter(
+                dask.dataframe.from_pandas(
+                    deserialize_arrow(base64.b64decode(self.doc.data_blob)),
+                    npartitions=self.doc.structure.macro.npartitions,
+                )
+            )
+
+    @property
+    def structure(self):
+        return DataFrameStructure.from_json(self.doc.structure)
+
+    @property
+    def metadata(self):
+        return self.doc.metadata
+
+    @dataframe_raise_if_inactive
+    def read(self, *args, **kwargs):
+        return self.dataframe_adapter.read(*args, **kwargs)
+
+    @dataframe_raise_if_inactive
+    def read_partition(self, *args, **kwargs):
+        return self.dataframe_adapter.read_partition(*args, **kwargs)
+
+    def microstructure(self):
+        return self.dataframe_adapter.microstructure()
+
+    def macrostructure(self):
+        return self.dataframe_adapter.macrostructure()
+
+    def put_data(self, body):
+        # Organize files into subdirectories with the first two
+        # charcters of the uid to avoid one giant directory.
+        path = self.directory / self.doc.uid[:2] / self.doc.uid
+        path.parent.mkdir(parents=True, exist_ok=True)
+
+        dataframe = deserialize_arrow(body)
+
+        dataframe.to_parquet(path)
+        self.collection.update_one(
+            {"uid": self.doc.uid},
+            {
+                "$set": {
+                    "data_url": f"file://localhost/{str(path).replace(os.sep, '/')}"
+                }
+            },
+        )
+
+
+class MongoAdapter(collections.abc.Mapping, IndexersMixin):
+    structure_family = "node"
+    include_routers = [router]
+    query_registry = QueryTranslationRegistry()
+    register_query = query_registry.register
+
+    def __init__(
+        self,
+        *,
+        database,
+        directory,
+        queries=None,
+        sorting=None,
+        metadata=None,
+        principal=None,
+        access_policy=None,
+    ):
+        self.database = database
+        self.collection = database["reconstructions"]
```

Comment:
maybe just take the collection object in the constructor instead of the database client that way you don't have to decide on a fixed collection name.

Reply:
@kleinhenz Thank you for catching this. That was a total miss on my part.
@danielballan, do you thin it a good idea if we add this option to the config file?

Reply:
I updated this line and now, it uses a collections name as `nodes`